exist-core/src/main/java/org/exist/xquery/AbstractFLWORClause.java

```diff
@@ -41,14 +41,10 @@ public AbstractFLWORClause(XQueryContext context) {
     }
 
     @Override
-    public LocalVariable createVariable(final String name) throws XPathException {
-        try {
-            final LocalVariable var = new LocalVariable(QName.parse(context, name, null));
-            firstVar = var;
-            return var;
-        } catch (final IllegalQNameException e) {
-            throw new XPathException(this, ErrorCodes.XPST0081, "No namespace defined for prefix " + name);
-        }
+    public LocalVariable createVariable(final QName name) throws XPathException {
+        final LocalVariable var = new LocalVariable(name);
+        firstVar = var;
```

Comment:
return var immediately?

Reply:
I think `firstVar` has to be set, as the variables kind of form a linked list of sorts in the state of the AbstractFLWORClause